test/timezones/rounding.jl

```diff
@@ -0,0 +1,234 @@
+    @test round(zdt, Dates.Hour) == ZonedDateTime(2016, 2, 5, 13, tz)
+    @test round(zdt, Dates.Minute) == ZonedDateTime(2016, 2, 5, 13, 10, tz)
+    @test round(zdt, Dates.Second) == ZonedDateTime(2016, 2, 5, 13, 10, 21, tz)
+end
+
+##########################
+# DST TRANSITION FORWARD #
+##########################
+
+# Test rounding over spring transition (missing hour). FixedTimeZones have no transitions,
+# but ZonedDateTimes with VariableTimeZones will round in their current (fixed) time zone
+# and then adjust to the new time zone if a transition has occurred (DST, for example).
+
+# Test rounding backward, toward the missing hour.
+
+dt = DateTime(2016, 3, 13, 3, 15)             # 15 minutes after transition
+
+zdt = ZonedDateTime(dt, fixed)
+@test floor(zdt, Dates.Day) == ZonedDateTime(2016, 3, 13, fixed)
+@test floor(zdt, Dates.Hour(2)) == ZonedDateTime(2016, 3, 13, 2, fixed)
+
+for tz in [winnipeg, st_johns]
+    zdt = ZonedDateTime(dt, tz)
+    @test floor(zdt, Dates.Day) == ZonedDateTime(2016, 3, 13, tz)
+    @test floor(zdt, Dates.Hour(2)) == ZonedDateTime(2016, 3, 13, 1, tz)
+end
+
+# Test rounding forward, toward the missing hour.
+
+dt = DateTime(2016, 3, 13, 1, 55)             # 5 minutes before transition
+
+zdt = ZonedDateTime(dt, fixed)
+@test ceil(zdt, Dates.Day) == ZonedDateTime(2016, 3, 14, fixed)
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(2016, 3, 13, 2, fixed)
+@test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(2016, 3, 13, 2, fixed)
+@test round(zdt, Dates.Minute(30)) == ZonedDateTime(2016, 3, 13, 2, fixed)
+
+for tz in [winnipeg, st_johns]
+    zdt = ZonedDateTime(dt, tz)
+
+    @test ceil(zdt, Dates.Day) == ZonedDateTime(2016, 3, 14, tz)
+    @test ceil(zdt, Dates.Hour) == ZonedDateTime(2016, 3, 13, 3, tz)
+    @test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(2016, 3, 13, 3, tz)
+    @test round(zdt, Dates.Minute(30)) == ZonedDateTime(2016, 3, 13, 3, tz)
+end
+
+# Test rounding from "midday".
+
+dt = DateTime(2016, 3, 13, 12)                # Noon on day of transition
+
+# Noon is the middle of the day, and ties round up by default.
+@test round(ZonedDateTime(dt, fixed), Dates.Day) == ZonedDateTime(2016, 3, 14, fixed)
+
+# Noon isn't the middle of the day, as 2:00 through 2:59:59.999 are missing in these zones.
+@test round(ZonedDateTime(dt, winnipeg), Dates.Day) == ZonedDateTime(2016, 3, 13, winnipeg)
+@test round(ZonedDateTime(dt, st_johns), Dates.Day) == ZonedDateTime(2016, 3, 13, st_johns)
+
+###########################
+# DST TRANSITION BACKWARD #
+###########################
+
+# Test rounding over autumn transition (additional, ambiguous hour).
+
+# Test rounding backward, toward the ambiguous hour.
+
+dt = DateTime(2015, 11, 1, 2, 15)             # 15 minutes after ambiguous hour
+
+zdt = ZonedDateTime(dt, fixed)
+@test floor(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, fixed)
+@test floor(zdt, Dates.Hour(3)) == ZonedDateTime(2015, 11, 1, fixed)
```

Comment:
Maybe worth explaining why you're flooring to an odd hour and a quick reminder of how this works.